frontend/src/tests/UsernameInput.spec.ts

```diff
@@ -0,0 +1,171 @@
+    let verifyButtonEl: HTMLButtonElement;
+
+    let cardUsernameEl: HTMLSpanElement;
+    let verifiedUserIdentity
+
+    const props = {
+        domain: UserType.I,
+        userId: '',
+        fieldId: 'userId',
+    }
+
+    const newProps = {
+        domain: UserType.B,
+        userId: 'newUserId',
+        fieldId: 'newFieldId',
+    };
+
+    const newValue = 'testIdir';
+
+    beforeEach(async () => {
+        wrapper = mount(UserNameInput, {
+            props,
+            global: {
+                plugins: [PrimeVue],
+            },
+        });
+
+        usernameInputText = wrapper.find('#userIdInput');
+        usernameInputTextEl = usernameInputText.element as HTMLInputElement;
+        verifyButton = wrapper.find("[data-target-btn='verifyIdir']");
+        verifyButtonEl = verifyButton.element as HTMLButtonElement;
+    });
+
+    afterEach(() => {
+        vi.clearAllMocks();
+        wrapper.unmount();
+    });
+
+    it('Should change usernameInput value', async () => {
+        await usernameInputText.setValue(newValue);
+        expect(usernameInputTextEl.value).toBe(newValue);
+    });
+
+    it('Should receive the correct prop', async () => {
+        //default props
+        expect(wrapper.props()).toEqual(props);
+
+         await wrapper.setProps(newProps);
+         expect(wrapper.props()).toEqual(newProps);
+         expect(wrapper.props()).not.toEqual(props)
+    });
+
+    it('Should call and emit correct value' , async () => {
+        await usernameInputText.setValue(newValue);
+        emitChange = wrapper.emitted('change');
+        expect(wrapper.emitted('change')).toBeTruthy();
+        // test the given parameters when emitChange has been called
+        // i.e. emitChange = [ [ 'B' ] ]
+        expect(emitChange![0][0]).toEqual(newValue);
+
+        emitSetVerifyResult = wrapper.emitted('setVerifyResult')
+        await wrapper.setProps(newProps);
+        await verifyButton.trigger('click');
+
+        expect(wrapper.emitted('setVerifyResult')).toBeTruthy();
+        expect(emitSetVerifyResult![1][0]).toEqual(true);
+    });
+
+    it('Should enable verify btn when username is inputted', async () => {
+        // button starts as disabled
+        expect((verifyButtonEl).disabled).toBe(true)
+        expect(verifyButton.classes('p-disabled')).toBe(true);
+
+        await wrapper.setProps({userId: newProps.userId })
+
+        expect((verifyButtonEl).disabled).toBe(false)
+        expect(verifyButton.classes('p-disabled')).toBe(false);
+    });
+
+    it('Should show loading on the verify btn while we call the api' , async () => {
+        vi.spyOn(
+            AppActlApiService.idirBceidProxyApi,
+            'idirSearch',
+        ).mockImplementation(async () => {
+            setLoadingState(true);
+            return userInputMock();
+        });
+
+        await usernameInputText.setValue(newValue);
+        expect(usernameInputTextEl.value).toBe(newValue);
```

Comment:
maybe no need to set value here, cause only set user id props can enable the button. But we do want to check loading is false, and button text is not loading before trigger the click

Reply:
I'll refactor it